lightning/src/chain/channelmonitor.rs

```diff
@@ -2196,12 +2222,43 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 						if per_commitment_points.0 == commitment_number + 1 { Some(point) } else { None }
 					} else { None };
 				if let Some(per_commitment_point) = per_commitment_point_option {
+					if let Some(transaction) = tx {
+						let revokeable_p2wsh_opt =
+							if let Ok(revocation_pubkey) = chan_utils::derive_public_revocation_key(
+								&self.secp_ctx, &per_commitment_point, &self.holder_revocation_basepoint)
+							{
+								if let Ok(delayed_key) = chan_utils::derive_public_key(&self.secp_ctx,
+									&per_commitment_point,
+									&self.counterparty_commitment_params.counterparty_delayed_payment_base_key)
+								{
+									Some(chan_utils::get_revokeable_redeemscript(&revocation_pubkey,
+										self.counterparty_commitment_params.on_counterparty_tx_csv,
+										&delayed_key).to_v0_p2wsh())
+								} else {
+									debug_assert!(false, "Failed to derive a delayed payment key for a commitment state we accepted");
+									None
+								}
+							} else {
+								debug_assert!(false, "Failed to derive a delayed payment key for a commitment state we accepted");
```

Comment:
I think this should be "Failed to derive a revocation payment key" instead of delayed?